StandaloneMmPkg/Core/Dispatcher.c

```diff
@@ -185,6 +185,47 @@ MmLoadImage (
   DriverEntry->ImageBuffer     = DstBuffer;
   DriverEntry->NumberOfPage    = PageCount;
 
+  Status = MmAllocatePool (
+             EfiRuntimeServicesData,
+             sizeof (EFI_LOADED_IMAGE_PROTOCOL),
+             (VOID **)&DriverEntry->LoadedImage
+             );
+  if (EFI_ERROR (Status)) {
+    MmFreePages (DstBuffer, PageCount);
+    return Status;
+  }
+
+  ZeroMem (DriverEntry->LoadedImage, sizeof (EFI_LOADED_IMAGE_PROTOCOL));

```

Comment:
Can we avoid allocating memory for DriverEntry->LoadedImage by adding "EFI_LOADED_IMAGE_PROTOCOL LoadedImage" field to DriverEntry structure?